### ISSUE TITLE
How to implement transfer learning / adaptation for tts

### ISSUE THREAD

**Issue by potato-inoue** [espnet/espnet#1208]
Now I try to implement speaker adaptation toward pre-trained model.
I added `torch_load(args.model, model)` into `espnet/tts/pytorch_backend/tts.py`. ([Codes](https://github.com/potato-inoue/espnet/commit/ec8d4fb8b73ff9b60dc66de9145dd1f3392d9b8f#diff-22acf0acda9e4fc1f05638b07a29baaa))
**But the loss became the same outline** even if `transformer-lr` decreased. ([Results](https://potato-inoue.github.io/Homepage/progress_report/Demo190911/#speaker-1089))
\* `encoder_alpha` & `decoder_alpha` changed by decreasing `transformer-lr`.

Could you give me some advise?
e.g., adding some codes, changing some configs, ...

[Adaptation condition]
Pre-trained model: [libritts.transformer.v1](https://github.com/espnet/espnet/blob/master/egs/libritts/tts1/RESULTS.md#v050-first-multi-speaker-transformer-1024-pt-window--256-pt-shift--x-vector-with-add-integration--fast-gl-64-iters)
Adaptation data: test_clean (open speaker for training)
Adaptation config: 
- `transformer-lr: 1` -> `transformer-lr: 1e-8`
- `epochs: 100` -> `epochs: 2`
- others becomes the same setting

[key wards]
transfer learning, speaker adaptation, fine-tuning

**Comment by r9y9** [comment]
Why do you call `model.eval()` right after loading the model parameters?  Is it intentional? 

What about trying single-speaker model adaptation first? Multi-speaker models are generally more difficult to train.

**Comment by kan-bayashi** [comment]
As @r9y9 said, only around 30 minutes speech is enough to fine-tune single speaker model (But I'm not sure it works with the different gender model).
I have no experience of the transformer based model adaptation, but I doubt that initial lr is too small. 
Did you already try the other lr settings or have any references?

**Comment by potato-inoue** [comment]
@r9y9 , @kan-bayashi 
I'm trying single-speaker model adaptation. I will show the result later.

>Did you already try the other lr settings or have any references?

I tried 9 kinds of lr ( 1e0 to 1e-8) in multi-speaker model adaptation. I have no reference at present.

**Comment by kan-bayashi** [comment]
And I heard that libritts includes pose in speech without any corresponding text.
This causes training failure.
To avoid this issue, VAE-based tacotron is needed.
https://arxiv.org/pdf/1810.07217.pdf

If I have a time, I will try to implement this model.

**Comment by r9y9** [comment]
That sounds really great 👍 

I noticed that @potato-inoue did trickly preprocessing for removing poses (correct me if I'm wrong) in #1056 (https://github.com/espnet/espnet/blob/a2181ad10929ae980c228f40533defa6904d9db0/egs/blizzard17/tts1/local/make_lab_wo_sil.py#L168-L181), but if we have GMVAE-like latent model, the model would be able to handle variations that are not explicitly shown in the input text (e.g. pause, speaking-style, emotion, etc).

**Comment by potato-inoue** [comment]
VAE or GMVAE is very attractive!
Now I try to select speaker that includes no-/low-pose in speech without any corresponding text.

**Comment by potato-inoue** [comment]
I checked that single-speaker model adaptation is running. `model.eval()` was the reason to not train. 
I have no pre-process for pose above you mentioned. Multi-speaker model has not checked yet.
- model: ljspeech.transformer.v1
- adapted speaker: speaker 237 (Female) from test_clean
<img src="https://user-images.githubusercontent.com/38646618/65694847-7481af00-e0b1-11e9-9851-e680d68b4882.png" width="320px">
<img src="https://user-images.githubusercontent.com/38646618/65694861-7d728080-e0b1-11e9-8588-23848fe2332b.png" width="320px">
<img src="https://user-images.githubusercontent.com/38646618/65694873-81060780-e0b1-11e9-8139-c5abc61014bf.png" width="320px">
<img src="https://user-images.githubusercontent.com/38646618/65694884-8400f800-e0b1-11e9-8da2-dfa4153998f1.png" width="320px">
